### PR TITLE
Feature/cdap 4715 tag explore for streams

### PR DESCRIPTION
**Pull request by yaojiefeng**
JIRA: https://issues.cask.co/browse/CDAP-4715
Build: http://builds.cask.co/browse/CDAP-DUT4716-2

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/StreamSystemMetadataWriter.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
  */
 public class StreamSystemMetadataWriter extends AbstractSystemMetadataWriter {
 
+  public static final String EXPLORE_TAG = "explore";
```
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/StreamSystemMetadataWriter.java]: replace existing usages of "explore" w/ this constant also?

Comment (yaojiefeng) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/StreamSystemMetadataWriter.java]: The other usages of "explore" are in DatasetSystemMetadataWriter, and that class has its own const EXPLORE_TAG, should we combine these two constants?

Comment (anwar6953) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/StreamSystemMetadataWriter.java]: yea, if easily possible

File: cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java [hunk 2]
```diff
@@ -94,6 +94,7 @@
   private final Id.Stream nonExistingStream = Id.Stream.from("blah", "mystream");
   private final Id.Stream.View nonExistingView = Id.Stream.View.from(nonExistingStream, "myView");
   private final Id.Artifact nonExistingArtifact = Id.Artifact.from(Id.Namespace.from("blah"), "art", "1.0.0");
+  private final String EXPLORE_TAG = "explore";
```
Comment (anwar6953) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/MetadataHttpHandlerTestRun.java]: Can't we use `AbstractSystemMetadataWriter.EXPLORE_TAG`

**Comment by albertshau** [comment]
lgtm

**Comment by albertshau** [comment]
are there plans to add this tag to streams during the upgrade process? Or will it only be there for streams created in 4.0.0 and up?

**Comment by yaojiefeng** [comment]
I think https://issues.cask.co/browse/CDAP-4708 already implemented this.  In the upgrade process, the tag will be added using ExistingEntitySystemMetadataWriter. @albertshau

**Comment by albertshau** [comment]
ah ok. lgtm

**Comment by yaojiefeng** [comment]
Build passed, merged

**Pull request merged**